vrk-kpa/suomifi-design-system
# [Fix] Upgrade vulnerable deps and other deps to be up to date.

## Description

- Upgrade vulnerable dependencies
- Update dependencies to latest available version
- Fix favicon missing with Firefox (and possibly other browsers)
  - Source image used was too small and autogeneration of the different sizes was not able to get good quality result before
- Change to use `suomifi-ui-components` **SkipLink** instead of the custom BypassLink implementation.
- Add typings and return types here and there
- Update Gatsby from v2 to **v4**
- Remove custom polyfills as they were not working anymore and by default Gatsby should already handle polyfills.
  - IE11 support is now dropped and per discussion there is no need to get it back.
- Update eslint dependencies
- Fix eslint configs
- Update prettier to use `bracketSameLine` because of deprecation of `jsxBracketSameLine`
- Update `husky` config for **pre-commit** and **pre-push** hooks
- Update `lint-staged` to check also **.tsx** files


## Known issues

~~Would be nice when reviewing the PR that these issues is looked by the reviewers to get them solved.~~

### Margin at breakpoint > 992px
~~At certain breakpoint, 992px, when increasing the width of browser margin is not correctly set to page if compared to current `develop` branch of the [DS-site](https://vrk-kpa.github.io/suomifi-design-system/fi/components/).~~
~~Issue might be in the main content of those problematic pages, **few** examples are~~

~~- components/~~
~~- components/modal/~~
~~- components/textarea/~~

@valger11 solved the issue in this commit 9b28da1334ac6ce09e0d47190afc9082c051779e
